src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1341,7 +1343,7 @@ private async ValueTask<EventProcessedState> TryProcessSymbolDeclaredAsync(Symbo
                 var skipDeclarationAnalysis = AnalysisScope.ShouldSkipDeclarationAnalysis(symbol);
                 var hasPerSymbolActions = AnalyzerActions.SymbolStartActionsCount > 0 && (!skipSymbolAnalysis || !skipDeclarationAnalysis);
 
-                AnalyzerActions perSymbolActions = hasPerSymbolActions ?
+                AnalyzerActions? perSymbolActions = hasPerSymbolActions ?
```

Comment:
Consider making this not nullable, and just use `Diagnostics.AnalyzerActions.Empty` for the false case, to remove the multiple `GetValueOrDefault` calls below.

Reply:
@sharwell this can still be simplified by simply using `AnalyzerActions.Empty` as the false case here, and removing the coalescing entirely.

---
In reply to: [362572645](https://github.com/dotnet/roslyn/pull/40691#discussion_r362572645) [](ancestors = 362572645)

Reply:
The code now uniformly uses `AnalyzerActions.Empty` (removed uses of both `null` and `default(AnalyzerActions)`.